frontend/iframe/viewmodels/RootViewModel.ts

```diff
@@ -112,8 +117,22 @@ export class RootViewModel extends ViewModel<SegmentType, Options> {
                 void this._showPicker();
             }
         } else if (sessionId) {
-            if (this.singleRoomMode) {
-                this.navigation.push("room", this.platform.config.roomId);
+            if (this._singleRoomIdOrAlias && !this._resolvedSingleRoomId) {
+                // We're in single-room mode but haven't resolved the room alias yet.
+                try {
+                    this._resolvedSingleRoomId = await this.resolveRoomAlias(sessionId, this._singleRoomIdOrAlias);
+                } catch (error) {
+                    // Something went wrong when navigating to the room.
+                    // We swallow the error and fallback to non-single-room mode.
```

Comment:
This is a good point, and I tend to agree we should probably not even render the client if we fail to resolve the room. Additionally, even if a room id was specified instead of an alias, the room might not exist. In short, it is not possible to guarantee that room exists nor that it has the right properties for the user to access it (e.g. rooms needs to be public, or be part of a Matrix Space the user has access to.

Since this issue was already present before this PR, my suggestion would be to tackle it later. I opened #181 to track it.

Reply:
Sure lets continue that in #181